framework/src/bcs/ADNodalBC.C

```diff
@@ -87,12 +89,15 @@ template <typename T>
 void
 ADNodalBCTempl<T>::computeJacobian()
 {
+  const std::vector<dof_id_type> & cached_rows = _var.dofIndices();
+  if (cached_rows.empty())
+    return;
```

Comment:
why do we need this now?

Reply:
It was potential landmine waiting to happen. We return early from residual evaluation if there are no dof indices, but we were not for the Jacobian